pkg/dockerregistry/server/signaturedispatcher.go

```diff
@@ -82,9 +82,9 @@ func (s *signatureHandler) Put(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	client, ok := userClientFrom(s.ctx)
-	if !ok {
-		s.handleError(s.ctx, errcode.ErrorCodeUnknown.WithDetail("unable to get origin client"), w)
+	client, err := RegistryClientFrom(s.ctx).Client()
```

Comment:
before we let the user create signatures via the registry, verifyImageSignatureAccess ensures they have `create imagesignatures` permission, so using the user client here would work. I'm not actually sure whether the registry has `create imagesignatures` permission by default